application/src/main/java/org/thingsboard/server/service/cloud/CloudManagerService.java

```diff
@@ -290,13 +297,20 @@ private void initAndUpdateEdgeSettings(EdgeConfiguration edgeConfiguration) thro
         edgeRpcClient.sendSyncRequestMsg(this.currentEdgeSettings.isFullSyncRequired());
         this.syncInProgress = true;
 
-        cloudEventService.saveEdgeSettings(tenantId, this.currentEdgeSettings);
+        edgeSettingsService.saveEdgeSettings(tenantId, this.currentEdgeSettings);
 
         saveOrUpdateEdge(tenantId, edgeConfiguration);
 
         updateConnectivityStatus(true);
 
         initialized = true;
+
+        if (queueType.equals("kafka") && !isCloudEventSync) {
+            cloudEventSync.init(tenantId);
+            cloudEventSync.cloudEventSync();
+            cloudEventSync.cloudEventTsSync();
+            isCloudEventSync = true;
```

Comment:
I think it's too overhead. Introduce method - processCloudEventMigration or similar where you will do the logic inside in class, that you need. 

queueType.equals("kafka") or use optional bean or check if it's exists - as you wish